robosat/osm/road.py

```diff
@@ -0,0 +1,160 @@
+            "unclassified",
+            "residential",
+            "service",
+            "motorway_link",
+            "trunk_link",
+            "primary_link",
+            "secondary_link",
+            "tertiary_link"
+        }
+
+    highway_attributes = {
+        'motorway': {
+            'lanes': 4,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.75,
+            'right_hard_shoulder_width': 3.0
+        },
+        'trunk': { 
+            'lanes': 3,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.75,
+            'right_hard_shoulder_width': 3.0
+        },
+        'primary': { 
+            'lanes': 2,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.50,
+            'right_hard_shoulder_width': 1.50
+        },
+        'secondary': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'tertiary': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'unclassified': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        },
+        'residential': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'service': { 
+            'lanes': 1,
+            'lane_width': 3.00,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        },
+        'motorway_link': { 
+            'lanes': 2,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.75,
+            'right_hard_shoulder_width': 3.00
+        },
+        'trunk_link': { 
+            'lanes': 2,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.50,
+            'right_hard_shoulder_width': 1.50
+        },
+        'primary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'secondary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'tertiary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        }
+    }
+
+    EARTH_MEAN_RADIUS = 6371004.0
+
+    def __init__(self):
+        super().__init__()
+        self.features = []
+
+    def way(self, w):
+        if "highway" not in w.tags:
+            return
+
+        if w.tags["highway"] not in self.road_filter():
```

Comment:
I think we should couple the road filter and the highway attributes. What about removing the explicit road filter and constructing it from the highway attributes dict? As in

```python
road_filter = set(highway_attributes.keys())
```

(not here in the way function though but once at the top)

Reply:
Yes, it woud be more elegant to only code the highway attributes once.